lib/services/settings/customize_service.dart

```diff
@@ -0,0 +1,285 @@
+  //BuildContext get context => null;
+
+
+  bool isCurrentlyDarkMode(BuildContext context) {
+  return MediaQuery.of(context).platformBrightness == Brightness.dark;
+}
+
+Future<List<String>> fetchGenresFromAPI(bool isMounted) async {
+    try {
+      var url =
+          Uri.parse('https://api.rawg.io/api/genres?key=${globals.apiKey}');
+      var response = await http.get(url);
+      if (response.statusCode == 200) {
+        var decoded = json.decode(response.body);
+        if (isMounted) {
+          return (decoded['results'] as List)
+                .map((genre) => genre['name'].toString())
+                .toList();
+        }
+        else{
+          return [];
+        }
+
+      } else {
+        //print("Error fetching genres: ${response.statusCode}");
+        return [];
+      }
+    } catch (e) {
+      //print("Error fetching genres: $e");
+    }
+    return [];
+  }
+
+  Future<List<String>> fetchTagsFromAPI(bool isMounted) async {
+    try {
+      if (isMounted) {
+        var url =
+            Uri.parse('https://api.rawg.io/api/tags?key=${globals.apiKey}');
+        var response = await http.get(url);
+        if (response.statusCode == 200) {
+          var decoded = json.decode(response.body);
+          return (decoded['results'] as List)
+                .map((tag) => tag['name'].toString())
+                .toList();
+        } else {
+          return [];
+        }
+      }
+      else{
+        return [];
+      }
+    } catch (e) {
+      //throw ("Error fetching interest tags: $e");
+    }
+    return [];
+  }
+
+  Future<List<List<String>>> fetchUserSelectionsFromDatabase() async {
+    try {
+      final FirebaseAuth auth = FirebaseAuth.instance;
+      final currentUser = auth.currentUser;
+
+      if (currentUser != null) {
+        final db = FirebaseFirestore.instance;
+        final profileDocRef =
+            db.collection("profile_data").doc(currentUser.uid);
+
+        final docSnapshot = await profileDocRef.get();
+        if (docSnapshot.exists) {
+          List<List<String>> customizeData=[];
+          final data = docSnapshot.data();
+          final genres = List<String>.from(data?["genre_interests_tags"] ?? []);
+          final age = List<String>.from(data?["age_rating_tags"] ?? []);
+          final interests =
+              List<String>.from(data?["social_interests_tags"] ?? []);
+
+          StorageService storageService = StorageService();
+          String bannerDownloadUrl =
+              await storageService.getBannerUrl(currentUser.uid);
+          String profileDownloadUrl =
+              await storageService.getProfilePictureUrl(currentUser.uid);
+
+          List<String> pictures = [];
+          pictures.add(bannerDownloadUrl);
+          pictures.add(profileDownloadUrl);
+
+          customizeData.add(genres);
+          customizeData.add(age);
+          customizeData.add(interests);
+          customizeData.add(pictures);
+
+          
+          /*if (_isMounted) {
```

Comment:
Just a question, why did you comment this part out? I believe this is to fix a memory leak error that @pieryvdl fixed

Reply:
ya its still there but on the customize page